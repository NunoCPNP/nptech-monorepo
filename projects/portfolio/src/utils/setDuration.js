export const setDuration = (minutes) => {
  const duration = new Date()

  duration.setTime(duration.getTime() + minutes * 60 * 1000)

  return duration
}
