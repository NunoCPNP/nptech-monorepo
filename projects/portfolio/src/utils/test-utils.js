import { render } from '@testing-library/react'

import { ThemeProvider } from 'emotion-theming'
import { dark } from '../styles/themes'
import { ControllersProvider } from '../context/controllers'

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={dark}>
      <ControllersProvider>{children}</ControllersProvider>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
