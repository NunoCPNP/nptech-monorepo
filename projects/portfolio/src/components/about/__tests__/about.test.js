import { render } from '../../../utils/test-utils'

import About from '../index'

describe('<About /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<About />)

    const test = queryAllByTestId(/about-component/i)
    expect(test).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<About />)

    expect(asFragment(<About />)).toMatchSnapshot()
  })
})
