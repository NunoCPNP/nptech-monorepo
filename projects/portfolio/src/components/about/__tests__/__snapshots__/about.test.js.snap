// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> spec Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ur40l5"
    data-testid="about-component"
  >
    <div />
  </div>
</DocumentFragment>
`;
